src/app/modules/catalog/components/super-offers/super-offers.component.ts

```diff
@@ -0,0 +1,75 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    Component,
+    OnDestroy,
+    OnInit,
+    Renderer2,
+    ViewChild,
+} from '@angular/core';
+import { Store } from '@ngrx/store';
+import { CatalogState } from '../../state/catalog.state';
+import * as OffersActions from '../../state/actions/offers.actions';
+import { Subject } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
+import * as OffersSelectors from '../../state/selectors/offers.selectors';
+import { selectedTab, selectedTabChanged } from '../../catalog.constants';
+
+@Component({
+    selector: 'app-super-offers',
+    templateUrl: './super-offers.component.html',
+    styleUrls: ['./super-offers.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SuperOffersComponent implements OnInit, AfterViewInit, OnDestroy {
+    @ViewChild('tabGroup') tabGroup: {
+        _tabBodyWrapper: { nativeElement: string };
+    };
+    private readonly unsubscribe$ = new Subject();
+    private element: { nativeElement: string };
+
+    readonly offers$ = this.store
+        .select(OffersSelectors.selectOffers)
+        .pipe(takeUntil(this.unsubscribe$));
+
+    readonly isReadyToDisplay$ = this.offers$.pipe(
+        map(el => !!el),
+        takeUntil(this.unsubscribe$),
+    );
+
+    constructor(
+        private readonly store: Store<CatalogState>,
+        private readonly renderer: Renderer2,
+    ) {}
+
+    ngOnInit(): void {
+        this.store.dispatch(OffersActions.getOffers());
+    }
+
+    ngAfterViewInit(): void {
+        this.element = this.tabGroup._tabBodyWrapper;
+    }
+
+    addClass(className: string, element: { nativeElement: string }) {
```

Comment:
why you won't do as it done here https://material.angular.io/components/tabs/examples#tab-group-dynamic
please rework